snappy/build.go

```diff
@@ -391,7 +391,11 @@ func copyToBuildDir(sourceDir, buildDir string) error {
 
 		// handle dirs
 		if info.IsDir() {
-			return os.Mkdir(dest, info.Mode())
+			if err := os.Mkdir(dest, info.Mode()); err != nil {
+				return err
+			}
+			// ensure that premissions are preserved
+			return os.Chown(dest, int(info.Sys().(*syscall.Stat_t).Uid), int(info.Sys().(*syscall.Stat_t).Gid))
```

Comment:
I'd split uid/gid to separate variables for clarity. I'm still new to go and I'm not sure what info.Sys() does but can we perhaps call it once instead of twice and just recover both fields?